from PyQt5 import QtCore, QtGui, QtWidgets
#from PyQt5.QtWidgets import QWidget, QLabel, QVBoxLayout
#from PyQt5.QtWidgets import QMainWindow, QFrame, QVBoxLayout, QPushButton, QApplication
import sys
import BaseFunctions as b

class Ui_offensiveAttributesWidget(QtWidgets.QWidget):
    def setupUi(self, offensiveAttributesWidget):
        offensiveAttributesWidget.setObjectName("offensiveAttributesWidget")
        offensiveAttributesWidget.resize(271, 108)
        self.offensiveLabel = QtWidgets.QLabel(offensiveAttributesWidget)
        self.offensiveLabel.setGeometry(QtCore.QRect(5, 0, 47, 13))
        self.offensiveLabel.setObjectName("offensiveLabel")
        self.shotThreeLabel = QtWidgets.QLabel(offensiveAttributesWidget)
        self.shotThreeLabel.setGeometry(QtCore.QRect(10, 20, 55, 13))
        self.shotThreeLabel.setObjectName("shotThreeLabel")
        self.line = QtWidgets.QFrame(offensiveAttributesWidget)
        self.line.setGeometry(QtCore.QRect(56, 0, 206, 16))
        self.line.setFrameShape(QtWidgets.QFrame.HLine)
        self.line.setFrameShadow(QtWidgets.QFrame.Sunken)
        self.line.setObjectName("line")
        self.shotMediumLabel = QtWidgets.QLabel(offensiveAttributesWidget)
        self.shotMediumLabel.setGeometry(QtCore.QRect(10, 40, 55, 13))
        self.shotMediumLabel.setObjectName("shotMediumLabel")
        self.shotCloseLabel = QtWidgets.QLabel(offensiveAttributesWidget)
        self.shotCloseLabel.setGeometry(QtCore.QRect(10, 60, 55, 13))
        self.shotCloseLabel.setObjectName("shotCloseLabel")
        self.shootInTrafficLabel = QtWidgets.QLabel(offensiveAttributesWidget)
        self.shootInTrafficLabel.setGeometry(QtCore.QRect(10, 80, 55, 13))
        self.shootInTrafficLabel.setObjectName("shootInTrafficLabel")
        self.shotInTrafficAttributeLabel = QtWidgets.QLabel(offensiveAttributesWidget)
        self.shotInTrafficAttributeLabel.setGeometry(QtCore.QRect(70, 80, 20, 13))
        self.shotInTrafficAttributeLabel.setObjectName("shotInTrafficAttributeLabel")
        self.shotCloseAttributeLabel = QtWidgets.QLabel(offensiveAttributesWidget)
        self.shotCloseAttributeLabel.setGeometry(QtCore.QRect(70, 60, 20, 13))
        self.shotCloseAttributeLabel.setObjectName("shotCloseAttributeLabel")
        self.shotMediumAttributeLabel = QtWidgets.QLabel(offensiveAttributesWidget)
        self.shotMediumAttributeLabel.setGeometry(QtCore.QRect(70, 40, 20, 13))
        self.shotMediumAttributeLabel.setObjectName("shotMediumAttributeLabel")
        self.shotThreeAttributeLabel = QtWidgets.QLabel(offensiveAttributesWidget)
        self.shotThreeAttributeLabel.setGeometry(QtCore.QRect(70, 20, 20, 13))
        self.shotThreeAttributeLabel.setObjectName("shotThreeAttributeLabel")
        self.line_2 = QtWidgets.QFrame(offensiveAttributesWidget)
        self.line_2.setGeometry(QtCore.QRect(10, 36, 80, 3))
        self.line_2.setFrameShape(QtWidgets.QFrame.HLine)
        self.line_2.setFrameShadow(QtWidgets.QFrame.Sunken)
        self.line_2.setObjectName("line_2")
        self.line_3 = QtWidgets.QFrame(offensiveAttributesWidget)
        self.line_3.setGeometry(QtCore.QRect(10, 56, 80, 3))
        self.line_3.setFrameShape(QtWidgets.QFrame.HLine)
        self.line_3.setFrameShadow(QtWidgets.QFrame.Sunken)
        self.line_3.setObjectName("line_3")
        self.line_4 = QtWidgets.QFrame(offensiveAttributesWidget)
        self.line_4.setGeometry(QtCore.QRect(10, 76, 80, 3))
        self.line_4.setFrameShape(QtWidgets.QFrame.HLine)
        self.line_4.setFrameShadow(QtWidgets.QFrame.Sunken)
        self.line_4.setObjectName("line_4")
        self.line_5 = QtWidgets.QFrame(offensiveAttributesWidget)
        self.line_5.setGeometry(QtCore.QRect(10, 100, 250, 3))
        self.line_5.setFrameShape(QtWidgets.QFrame.HLine)
        self.line_5.setFrameShadow(QtWidgets.QFrame.Sunken)
        self.line_5.setObjectName("line_5")
        self.shootOffDribbleLabel = QtWidgets.QLabel(offensiveAttributesWidget)
        self.shootOffDribbleLabel.setGeometry(QtCore.QRect(100, 20, 55, 13))
        self.shootOffDribbleLabel.setObjectName("shootOffDribbleLabel")
        self.postHookLabel = QtWidgets.QLabel(offensiveAttributesWidget)
        self.postHookLabel.setGeometry(QtCore.QRect(100, 60, 55, 13))
        self.postHookLabel.setObjectName("postHookLabel")
        self.postFadeawayAttributeLabel = QtWidgets.QLabel(offensiveAttributesWidget)
        self.postFadeawayAttributeLabel.setGeometry(QtCore.QRect(160, 40, 20, 13))
        self.postFadeawayAttributeLabel.setObjectName("postFadeawayAttributeLabel")
        self.line_6 = QtWidgets.QFrame(offensiveAttributesWidget)
        self.line_6.setGeometry(QtCore.QRect(100, 76, 80, 3))
        self.line_6.setFrameShape(QtWidgets.QFrame.HLine)
        self.line_6.setFrameShadow(QtWidgets.QFrame.Sunken)
        self.line_6.setObjectName("line_6")
        self.postHookAttributeLabel = QtWidgets.QLabel(offensiveAttributesWidget)
        self.postHookAttributeLabel.setGeometry(QtCore.QRect(160, 60, 20, 13))
        self.postHookAttributeLabel.setObjectName("postHookAttributeLabel")
        self.shootOffDribbleAttributeLabel = QtWidgets.QLabel(offensiveAttributesWidget)
        self.shootOffDribbleAttributeLabel.setGeometry(QtCore.QRect(160, 20, 20, 13))
        self.shootOffDribbleAttributeLabel.setObjectName("shootOffDribbleAttributeLabel")
        self.lowPostOffenseLabel = QtWidgets.QLabel(offensiveAttributesWidget)
        self.lowPostOffenseLabel.setGeometry(QtCore.QRect(100, 80, 55, 13))
        self.lowPostOffenseLabel.setObjectName("lowPostOffenseLabel")
        self.line_7 = QtWidgets.QFrame(offensiveAttributesWidget)
        self.line_7.setGeometry(QtCore.QRect(100, 36, 80, 3))
        self.line_7.setFrameShape(QtWidgets.QFrame.HLine)
        self.line_7.setFrameShadow(QtWidgets.QFrame.Sunken)
        self.line_7.setObjectName("line_7")
        self.postFadeawayLabel = QtWidgets.QLabel(offensiveAttributesWidget)
        self.postFadeawayLabel.setGeometry(QtCore.QRect(100, 40, 55, 13))
        self.postFadeawayLabel.setObjectName("postFadeawayLabel")
        self.lowPostOffenseAttributeLabel = QtWidgets.QLabel(offensiveAttributesWidget)
        self.lowPostOffenseAttributeLabel.setGeometry(QtCore.QRect(160, 80, 20, 13))
        self.lowPostOffenseAttributeLabel.setObjectName("lowPostOffenseAttributeLabel")
        self.line_8 = QtWidgets.QFrame(offensiveAttributesWidget)
        self.line_8.setGeometry(QtCore.QRect(100, 56, 80, 3))
        self.line_8.setFrameShape(QtWidgets.QFrame.HLine)
        self.line_8.setFrameShadow(QtWidgets.QFrame.Sunken)
        self.line_8.setObjectName("line_8")
        self.consistencyAttributeLabel = QtWidgets.QLabel(offensiveAttributesWidget)
        self.consistencyAttributeLabel.setGeometry(QtCore.QRect(250, 20, 20, 13))
        self.consistencyAttributeLabel.setObjectName("consistencyAttributeLabel")
        self.consistencyLabel = QtWidgets.QLabel(offensiveAttributesWidget)
        self.consistencyLabel.setGeometry(QtCore.QRect(190, 20, 55, 13))
        self.consistencyLabel.setObjectName("consistencyLabel")
        self.line_9 = QtWidgets.QFrame(offensiveAttributesWidget)
        self.line_9.setGeometry(QtCore.QRect(190, 36, 80, 3))
        self.line_9.setFrameShape(QtWidgets.QFrame.HLine)
        self.line_9.setFrameShadow(QtWidgets.QFrame.Sunken)
        self.line_9.setObjectName("line_9")
        self.line_10 = QtWidgets.QFrame(offensiveAttributesWidget)
        self.line_10.setGeometry(QtCore.QRect(85, 15, 20, 81))
        self.line_10.setFrameShape(QtWidgets.QFrame.VLine)
        self.line_10.setFrameShadow(QtWidgets.QFrame.Sunken)
        self.line_10.setObjectName("line_10")
        self.line_11 = QtWidgets.QFrame(offensiveAttributesWidget)
        self.line_11.setGeometry(QtCore.QRect(175, 15, 20, 81))
        self.line_11.setFrameShape(QtWidgets.QFrame.VLine)
        self.line_11.setFrameShadow(QtWidgets.QFrame.Sunken)
        self.line_11.setObjectName("line_11")

        self.retranslateUi(offensiveAttributesWidget)
        QtCore.QMetaObject.connectSlotsByName(offensiveAttributesWidget)

    def retranslateUi(self, offensiveAttributesWidget):
        _translate = QtCore.QCoreApplication.translate
        offensiveAttributesWidget.setWindowTitle(_translate("offensiveAttributesWidget", "offensiveAttributesWidget"))
        self.offensiveLabel.setText(_translate("offensiveAttributesWidget", "Offensive"))
        self.shotThreeLabel.setText(_translate("offensiveAttributesWidget", "Shot 3pt"))
        self.shotMediumLabel.setText(_translate("offensiveAttributesWidget", "Shot Med"))
        self.shotCloseLabel.setText(_translate("offensiveAttributesWidget", "Shot Cls"))
        self.shootInTrafficLabel.setText(_translate("offensiveAttributesWidget", "In Traffic"))
        self.shotInTrafficAttributeLabel.setText(_translate("offensiveAttributesWidget", "99"))
        self.shotCloseAttributeLabel.setText(_translate("offensiveAttributesWidget", "99"))
        self.shotMediumAttributeLabel.setText(_translate("offensiveAttributesWidget", "99"))
        self.shotThreeAttributeLabel.setText(_translate("offensiveAttributesWidget", "99"))
        self.shootOffDribbleLabel.setText(_translate("offensiveAttributesWidget", "Off Dribble"))
        self.postHookLabel.setText(_translate("offensiveAttributesWidget", "PHook"))
        self.postFadeawayAttributeLabel.setText(_translate("offensiveAttributesWidget", "99"))
        self.postHookAttributeLabel.setText(_translate("offensiveAttributesWidget", "99"))
        self.shootOffDribbleAttributeLabel.setText(_translate("offensiveAttributesWidget", "99"))
        self.lowPostOffenseLabel.setText(_translate("offensiveAttributesWidget", "LP Offense"))
        self.postFadeawayLabel.setText(_translate("offensiveAttributesWidget", "PFadeaway"))
        self.lowPostOffenseAttributeLabel.setText(_translate("offensiveAttributesWidget", "99"))
        self.consistencyAttributeLabel.setText(_translate("offensiveAttributesWidget", "99"))
        self.consistencyLabel.setText(_translate("offensiveAttributesWidget", "Consistent"))
class OffensiveAttributesWidget(QtWidgets.QWidget):
    def __init__(self, parent=None):
        super(OffensiveAttributesWidget, self).__init__(parent)
        self.ui = Ui_offensiveAttributesWidget()
        self.ui.setupUi(self)

    # You can add methods to update this widget's UI based on data
    def update_shotThreeAttribute(self, value):
        self.ui.shotThreeAttributeLabel.setText(str(value))


class Ui_playerResultWidget(QtWidgets.QWidget):
    def setupUi(self, playerResultWidget):
        playerResultWidget.setObjectName("playerResultWidget")
        playerResultWidget.resize(313, 793)
        self.playerNameLabel = QtWidgets.QLabel(playerResultWidget)
        self.playerNameLabel.setGeometry(QtCore.QRect(50, -20, 211, 81))
        sizePolicy = QtWidgets.QSizePolicy(QtWidgets.QSizePolicy.Expanding, QtWidgets.QSizePolicy.Preferred)
        sizePolicy.setHorizontalStretch(0)
        sizePolicy.setVerticalStretch(0)
        sizePolicy.setHeightForWidth(self.playerNameLabel.sizePolicy().hasHeightForWidth())
        self.playerNameLabel.setSizePolicy(sizePolicy)
        font = QtGui.QFont()
        font.setFamily("Segoe UI")
        font.setPointSize(12)
        font.setBold(True)
        font.setWeight(75)
        self.playerNameLabel.setFont(font)
        self.playerNameLabel.setFrameShape(QtWidgets.QFrame.NoFrame)
        self.playerNameLabel.setFrameShadow(QtWidgets.QFrame.Sunken)
        self.playerNameLabel.setTextFormat(QtCore.Qt.PlainText)
        self.playerNameLabel.setScaledContents(False)
        self.playerNameLabel.setAlignment(QtCore.Qt.AlignCenter)
        self.playerNameLabel.setWordWrap(True)
        self.playerNameLabel.setObjectName("playerNameLabel")
        self.archetypeLabel = QtWidgets.QLabel(playerResultWidget)
        self.archetypeLabel.setGeometry(QtCore.QRect(68, 50, 171, 21))
        sizePolicy = QtWidgets.QSizePolicy(QtWidgets.QSizePolicy.Expanding, QtWidgets.QSizePolicy.Preferred)
        sizePolicy.setHorizontalStretch(0)
        sizePolicy.setVerticalStretch(0)
        sizePolicy.setHeightForWidth(self.archetypeLabel.sizePolicy().hasHeightForWidth())
        self.archetypeLabel.setSizePolicy(sizePolicy)
        font = QtGui.QFont()
        font.setFamily("Segoe UI")
        font.setPointSize(10)
        font.setBold(False)
        font.setItalic(True)
        font.setWeight(50)
        self.archetypeLabel.setFont(font)
        self.archetypeLabel.setFrameShape(QtWidgets.QFrame.NoFrame)
        self.archetypeLabel.setFrameShadow(QtWidgets.QFrame.Sunken)
        self.archetypeLabel.setTextFormat(QtCore.Qt.PlainText)
        self.archetypeLabel.setScaledContents(False)
        self.archetypeLabel.setAlignment(QtCore.Qt.AlignCenter)
        self.archetypeLabel.setWordWrap(True)
        self.archetypeLabel.setObjectName("archetypeLabel")
        self.rarityLabel = QtWidgets.QLabel(playerResultWidget)
        self.rarityLabel.setGeometry(QtCore.QRect(71, 70, 161, 21))
        sizePolicy = QtWidgets.QSizePolicy(QtWidgets.QSizePolicy.Expanding, QtWidgets.QSizePolicy.Preferred)
        sizePolicy.setHorizontalStretch(0)
        sizePolicy.setVerticalStretch(0)
        sizePolicy.setHeightForWidth(self.rarityLabel.sizePolicy().hasHeightForWidth())
        self.rarityLabel.setSizePolicy(sizePolicy)
        font = QtGui.QFont()
        font.setFamily("System")
        font.setPointSize(12)
        font.setBold(True)
        font.setItalic(False)
        font.setWeight(75)
        self.rarityLabel.setFont(font)
        self.rarityLabel.setFrameShape(QtWidgets.QFrame.NoFrame)
        self.rarityLabel.setFrameShadow(QtWidgets.QFrame.Sunken)
        self.rarityLabel.setTextFormat(QtCore.Qt.PlainText)
        self.rarityLabel.setScaledContents(False)
        self.rarityLabel.setAlignment(QtCore.Qt.AlignCenter)
        self.rarityLabel.setWordWrap(True)
        self.rarityLabel.setObjectName("rarityLabel")
        self.backstoryLabel = QtWidgets.QLabel(playerResultWidget)
        self.backstoryLabel.setGeometry(QtCore.QRect(20, 110, 281, 171))
        self.backstoryLabel.setAlignment(QtCore.Qt.AlignLeading|QtCore.Qt.AlignLeft|QtCore.Qt.AlignTop)
        self.backstoryLabel.setWordWrap(True)
        self.backstoryLabel.setObjectName("backstoryLabel")
        self.backstorySeparater = QtWidgets.QFrame(playerResultWidget)
        self.backstorySeparater.setGeometry(QtCore.QRect(20, 280, 281, 20))
        self.backstorySeparater.setFrameShape(QtWidgets.QFrame.HLine)
        self.backstorySeparater.setFrameShadow(QtWidgets.QFrame.Sunken)
        self.backstorySeparater.setObjectName("backstorySeparater")
        self.factionIcon = QtWidgets.QLabel(playerResultWidget)
        self.factionIcon.setGeometry(QtCore.QRect(20, 34, 71, 71))
        self.factionIcon.setText("")
        self.factionIcon.setPixmap(QtGui.QPixmap("FactionIcons/Sludgeservants.png"))
        self.factionIcon.setScaledContents(True)
        self.factionIcon.setObjectName("factionIcon")
        self.artifactIcon = QtWidgets.QLabel(playerResultWidget)
        self.artifactIcon.setGeometry(QtCore.QRect(220, 34, 71, 71))
        self.artifactIcon.setText("")
        self.artifactIcon.setPixmap(QtGui.QPixmap("FactionIcons/American Illegals.jpg"))
        self.artifactIcon.setScaledContents(True)
        self.artifactIcon.setObjectName("artifactIcon")
        self.infoTabs = QtWidgets.QTabWidget(playerResultWidget)
        self.infoTabs.setGeometry(QtCore.QRect(20, 300, 281, 471))
        self.infoTabs.setObjectName("infoTabs")
        self.attributesTab = QtWidgets.QWidget()
        self.attributesTab.setObjectName("attributesTab")
        self.primaryAttributesHolder = QtWidgets.QFrame(self.attributesTab)
        self.primaryAttributesHolder.setGeometry(QtCore.QRect(0, 0, 271, 150))
        self.primaryAttributesHolder.setFrameShape(QtWidgets.QFrame.StyledPanel)
        self.primaryAttributesHolder.setFrameShadow(QtWidgets.QFrame.Raised)
        self.primaryAttributesHolder.setObjectName("primaryAttributesHolder")
        self.secondaryAttributesHolder = QtWidgets.QFrame(self.attributesTab)
        self.secondaryAttributesHolder.setGeometry(QtCore.QRect(0, 110, 271, 108))
        self.secondaryAttributesHolder.setFrameShape(QtWidgets.QFrame.StyledPanel)
        self.secondaryAttributesHolder.setFrameShadow(QtWidgets.QFrame.Raised)
        self.secondaryAttributesHolder.setObjectName("secondaryAttributesHolder")
        self.tertiaryAttributesHolder = QtWidgets.QFrame(self.attributesTab)
        self.tertiaryAttributesHolder.setGeometry(QtCore.QRect(0, 220, 271, 108))
        self.tertiaryAttributesHolder.setFrameShape(QtWidgets.QFrame.StyledPanel)
        self.tertiaryAttributesHolder.setFrameShadow(QtWidgets.QFrame.Raised)
        self.tertiaryAttributesHolder.setObjectName("tertiaryAttributesHolder")
        self.otherAttributesHolder = QtWidgets.QFrame(self.attributesTab)
        self.otherAttributesHolder.setGeometry(QtCore.QRect(0, 330, 271, 108))
        self.otherAttributesHolder.setFrameShape(QtWidgets.QFrame.StyledPanel)
        self.otherAttributesHolder.setFrameShadow(QtWidgets.QFrame.Raised)
        self.otherAttributesHolder.setObjectName("otherAttributesHolder")
        self.infoTabs.addTab(self.attributesTab, "")
        self.statsTab = QtWidgets.QWidget()
        self.statsTab.setObjectName("statsTab")
        self.infoTabs.addTab(self.statsTab, "")

        self.retranslateUi(playerResultWidget)
        self.infoTabs.setCurrentIndex(0)
        QtCore.QMetaObject.connectSlotsByName(playerResultWidget)

    def retranslateUi(self, playerResultWidget):
        _translate = QtCore.QCoreApplication.translate
        playerResultWidget.setWindowTitle(_translate("playerResultWidget", "playerResultWidget"))
        self.playerNameLabel.setText(_translate("playerResultWidget", "Player Name"))
        self.archetypeLabel.setText(_translate("playerResultWidget", "Archetype"))
        self.rarityLabel.setText(_translate("playerResultWidget", "Rarity"))
        self.backstoryLabel.setText(_translate("playerResultWidget", "This is where the player\'s backstory will go. It will describe their origins, along with many interesting stories which led them to become the person they are today. These are generated by ChatGPT 4.0, and deliver esquisite tales which are faction-based and varied by a multitude of modifiers. So, sit right down and enjoy a heart-warming backstory about Spritopias newest basketball talent. This next sentence is purely to fill more space, as I am trying to see how much room I need to provide to create adequate space for these backstories, but also not too much to leave tons of blank, unusable space where other fun and interesting things can be put."))
        self.infoTabs.setTabText(self.infoTabs.indexOf(self.attributesTab), _translate("playerResultWidget", "Attributes"))
        self.infoTabs.setTabText(self.infoTabs.indexOf(self.statsTab), _translate("playerResultWidget", "Stats"))
# This is our new, more functional class that extends QWidget
class PlayerResultWidget(QtWidgets.QWidget):
    def __init__(self, parent=None):
        super(PlayerResultWidget, self).__init__(parent)
        # Initialize the UI
        self.ui = Ui_playerResultWidget()
        self.ui.setupUi(self)

        # Setup layouts for all holder widgets
        self.ui.primaryAttributesHolder.setLayout(QtWidgets.QVBoxLayout())
        self.ui.secondaryAttributesHolder.setLayout(QtWidgets.QVBoxLayout())
        self.ui.tertiaryAttributesHolder.setLayout(QtWidgets.QVBoxLayout())
        self.ui.otherAttributesHolder.setLayout(QtWidgets.QVBoxLayout())

        # Additional setup, connect signals, etc.
        # You can set this up here or in a separate method
        self.offensiveAttributesWidget = OffensiveAttributesWidget()
        self.ui.primaryAttributesHolder.layout().addWidget(self.offensiveAttributesWidget)

    def setPlayerName(self, name):
        self.ui.playerNameLabel.setText(name)


app = QtWidgets.QApplication([])

window = PlayerResultWidget()
window.show()
window.setPlayerName("John Doe")

app.exec_()